test/functional/client_side_encryption_corpus_tests.js

```diff
@@ -0,0 +1,381 @@
+'use strict';
+
+// The corpus test exhaustively enumerates all ways to encrypt all BSON value types. Note, the test data includes BSON binary subtype 4 (or standard UUID), which MUST be decoded and encoded as subtype 4. Run the test as follows.
+
+const fs = require('fs');
+const path = require('path');
+const EJSON = require('mongodb-extjson');
+const chai = require('chai');
+const expect = chai.expect;
+chai.config.includeStack = true;
+chai.config.showDiff = true;
+chai.config.truncateThreshold = 0;
+
+describe('Client Side Encryption Corpus', function() {
+  // See if we can run these tests
+  if (process.env.AWS_ACCESS_KEY_ID == null || process.env.AWS_SECRET_ACCESS_KEY == null) {
+    console.log('skipping Client Side Encryption Corpus tests due to lack of AWS credentials');
+    return;
+  }
+
+  let mongodbClientEncryption;
+  try {
+    mongodbClientEncryption = require('mongodb-client-encryption')(require('../../index'));
+  } catch (e) {
+    console.log(
+      'skipping Client Side Encryption Corpus tests due to inability to load mongodb-client-encryption'
+    );
+    return;
+  }
+
+  const corpusDir = path.resolve(__dirname, 'spec', 'client-side-encryption', 'corpus');
+  function loadCorpusData(filename) {
+    return EJSON.parse(fs.readFileSync(path.resolve(corpusDir, filename), { strict: true }));
```

Comment:
nit: if we're going to write utility functions like this, I would prefer not to use variables scoped outside of the function (`corpusDir`), but rather pass them in.

Reply:
not sure what you mean here. Are you saying you want:

```js
function loadCorpusData(filename) {
  return EJSON.parse(fs.readFileSync(path.resolve(__dirname, 'spec', 'client-side-encryption', 'corpus', filename)), { strict: true });
}
const corpusSchema = loadCorpusData('corpus-schema.json');
```

or do you want

```js
const corpusDir = path.resolve(__dirname, 'spec', 'client-side-encryption', 'corpus');
function loadCorpusData(dir, filename) {
  return EJSON.parse(fs.readFileSync(path.resolve(dir, filename), { strict: true }));
}

const corpusSchema = loadCorpusData(corpusDir, 'corpus-schema.json');
```

Reply:
both look file, but the latter is what I had in mind originally